### PR TITLE
Adds refreshed Cyrillic in the NotoSansRoman source

### PR DESCRIPTION
**Pull request by kalapi**
I have been asked by Ksenya to help with merging in the Cyrillic updates to Noto L.G.C.

My concerns at the moment are:

1. Because this update is only for the core Cyrillic set I need to ensure that the new kerning is integrated without error. From my perspective, this is a complex Q.A. task because the previous versions of the Cyrillic used the same kerning classes as the Latin for glyphs that shared similar shapes. Meanwhile, the new kerning uses Cyrillic-specific kerning class names. A visual diff of the full Cyrillic set maybe a good start but also HB-regression tests?

More soon...

### CODE REVIEW

**Comment by simoncozens** [comment]
If we can get the CI running on this PR, it will give you a diffenator2 report against the previous release. I'm not sure why it isn't running.

**Comment by samarskaya** [comment]
Super— thanks for the merging assist @kalapi. Looks like all is ready to be merged upstream.

**Comment by simoncozens** [comment]
> Looks like all is ready to be merged upstream.

Well, first we QA it. :-)

* In Serif, *all base Cyrillic glyphs seem to have lost their anchors*. We need top, centre and bottom anchors on everything.
* Locale substitutions not being made for Cyrillic marks (dieresiscomb.localCYR) - but do we even need them? They are not used in precomposed forms; e.g. ӓ uses the standard Latin dieresis.
* In the Sans and Serif, Latin U+025C REVERSED OPEN E (ɜ) has changed. Does that matter? I think I'm happy with it.
* In Serif, LATIN SMALL CAPITAL B has also changed, but I like it.
* In Sans, interpolation issues in щ, Ԯ:
lgc
* In Sans Italic, dots on Їј have subtle interpolation issues between light and regular weights.
* С in the Mono has an interpolation issue:
lgc

**Comment by samarskaya** [comment]
Perfect— thanks for this list. Will ping once updated.

**Comment by moyogo** [comment]
There’s a few Latin or Greek glyphs that use Cyrillic components that are now kerned with Cyrillic, they shouldn’t be kerned with Cyrillic (but really with Latin).
- /Btopbar is now kerned as En-cy and Be-cy in NotoSans, NotoSans-Italic, as En-cy in NotoSerif-Italic, not otherwise, it should kern with Latin or not as previously.
- /Et is now kerned as Ze-cy and Ve-cy in NotoSans-Italic, not otherwise, it should kern with Latin or not as previously.
- /Pamphyliandigamma is now kerned as En-cy and En-cy in Notosans, NotoSans-Italic, it should kern with Greek or not as previously
- /eopenreversed is now kerned as ze-cy and ve-cy in NotoSans, NotoSans-Italic, NotoSerif, it should kern with Latin or not as previously
- /eopenreversedhook is now kerned as ze-cy on the left in NotoSans-Italic
- /epsilonLunateReversedSymbol/scruple are now kerned as zhe-cy and o-cy in NotoSans, NotoSans-Italic
- /epsilonLunateSymbol is now kerned as o-cy and zhe-cy in NotoSans, NotoSans-Italic

/cyrillicbreve/cyrillicbreve.cap
- /cyrillicbreve.cap should be adjusted to look like the outline used in /Abreve-cy/Iebreve-cy and these should use it as a component as /Zhebreve-cy instead of an outline in NotoSans, NotoSans-Italic. 
- /cyrillicbreve should be adjusted to look like the outline used in /abreve-cy/iebreve-cy and these should use it as a component as /zhebreve-cy instead of an outline in NotoSans, NotoSans-Italic.
- /cyrillicbreve.cap doesn’t seem to be used in NotoSerif, either use in Abreve-cy/Iebreve-cy/Zhebreve-cy or drop it.
- /cyrillicbreve/cyrillicbreve.cap need top anchor in NotoSerif, NotoSerif-Italic and _top anchor in NotoSans, NotoSans-Italic, NotoSerif, NotoSerif-Italic, NotoSansMono.
- to add to the "All base glyphs should have anchors" /breve is used on the following and ӐӑГ̆г̆Ҕ̆ҕ̆ӖӗӁӂꚄ̆ꚅ̆ЙйӃ̆ӄ̆Ԛ̆ԛ̆Ө̆ө̆Ꚍ̆ꚍ̆ЎўХ̆х̆Ꚏ̆ꚏ̆Ч̆ч̆Ꚇ̆ꚇ̆Ҽ̆ҽ̆Џ̆џ̆Ш̆ш̆Ꚗ̆ꚗ̆Ы̆ы̆Э̆э̆Я̆я̆ 
- There should be some lookups for Cyrl in ccmp with `sub @Cyrillic brevecomb' by cyrillicbreve;` and `sub @UppercaseCyrillic cyrillicbreve' by cyrillicbreve.cap;`

> In the Sans and Serif, Latin U+025C REVERSED OPEN E (ɜ) has changed. Does that matter? I think I'm happy with it.

- U+025C REVERSED OPEN E (ɜ) and U+025B OPEN E (ɛ) should be consistant. The difference is notable in NotoSerif and NotoSerif-Italic where the terminals are heavier on /Ze-cy/ze-cy (Зз) and thus on the new /Ereversedopen/eopenreversed (Ɜɜ) in Thin weights compared to /Eopen/eopen (Ɛɛ). Note that /reversedze-cy (ԑ) is used as a component for /eopen (ɛ) so when /reversedze-cy (ԑ) changes so will /eopen (ɛ). This PR should not change the design of Latin ɛɜɝᴈᶓᵋᵌ or ƐꞫ yet. Decompose them if they are using Cyrillic glyphs that are changing as components. For NotoSans, NotoSans-Italic, NotoSansMono the difference is less noticeable and /eopen/reversedze-cy both use /epsilon as a component at the moment.
- Similarly /epsilonLunateReversedSymbol/epsilonLunateSymbol/scruple are being changed because they use /ereversed-cy/e-cy and have a notable difference with glyphs like /sigmaLunateReversedSymbol/sigmaLunateSymbol. Their design should not change in this PR and should be decomposed instead.
- /phi-latin was decomposed in NotoSerif to not change along with it previous component /ef-cy, which is the right approach. In NotoSansMono, /phi-latin still uses /ef-cy which has changed, in this case its a desirable improvement. NotoSans-Italic /phi-latin/phiSymbol/ef-cy seem to have the same outline, two should use on as a component.

> In Serif, LATIN SMALL CAPITAL B has also changed, but I like it.

- This is arguable. The new design of is better for Cyrillic. One would need to look at /Bsmall/Bbarredsmall with other Latin glyphs, specifically IPA or UPA glyphs or the whole set of Latin small capital letters ᴀᴀᴁʙᴃᴄᴅᴅᴆᴇᴇⱻꜰɢʛ𝼂ʜɪᵻᴊᴋ𝼐ʟᴌ𝼄ᴍɴᴎᴏɶᴐᴕᴘʀꭆᴙᴚʁꝶꜱᴛᴜᵾꟺᴠᴡʏᴢᴣ. /Psmall ᴘ seem noticeably narrow compared to /Bsmall ʙ now, maybe ʀᴙᴚʁꭆꝶ should be considered as well as they seem a bit narrower but they are sometimes inconsistant between themselves. In any case, the NotoSerif /Bsmall/Bbarredsmall are unchanged and should be adjusted in a future update to follow this change if this is it.

**Comment by moyogo** [comment]
- Regarding #149 and ѣꭠ /yat-cy/sakhayat in NotoSans-Italic and NotoSerif-Italic, there are possible shapes for ѣ /yat-cy distinct from ꭠ /sakhayat.
See for example the italic:

or the roundhands:



- The ᲇ /yattall-cy should also be modified in NotoSerif-Italic as it currently has the same shape as /yat-cy/sakhayat and in  in NotoSans-Italic as it is slanted.

**Comment by kalapi** [comment]
Hello @moyogo, @simoncozens, and @samarskaya 

I'm starting to incorporate the changes based on Ksenya's new sources. Adding notes as I go below:
#### Noto Sans Roman
- [ ] Merge new changes (interpolation issues)
- [ ] Retain anchors in all updated bases
- [ ] Decompose non-Cyrillic glyphs and revert to pre-merge data.
#### Noto Sans Italic
#### Noto Sans Mono

**Comment by samarskaya** [comment]
Thank you all— 

Correct re ѣꭠ and even Ӑӑ being out-of-scope for this round, but def happy to make sure that the updated correct Cyrillic breve is on all things it’s easy to anchor onto. 

And good to de-couple different script components and kern groups for future sanity wherever we find them. :)

**Comment by simoncozens** [comment]
> This is arguable. The new design of is better for Cyrillic. One would need to look at /Bsmall/Bbarredsmall with other Latin glyphs, specifically IPA or UPA glyphs or the whole set of Latin small capital letters 

Yeah, you're right. Ideally this should not touch the Latin at all.

**Comment by kalapi** [comment]
I've updated the sources to address all the above issues. Can you perform another round of Q.A.? Thanks!

File: sources/NotoSansMono.glyphspackage/UIState.plist [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+displayStrings = (
+"⌠",
+"∌∉⋢⋣⊄⊃⊅"
```
Comment (kalapi) [inline sources/NotoSansMono.glyphspackage/UIState.plist]: Ha! I thought that I had caused this and fixed it in my sources. I'm sorry if that fix caused you trouble!

**Comment by simoncozens** [comment]
More QA on the latest sources:

* Sans: Interpolation issues with ё dots, Э crossbar
* Serif: Interpolation issues with ї dots

(I recommend using `fonttools varLib.interpolatable` or fontbakery to check these issues.)

Otherwise looking good. And in fact, I can fix these myself.

**Pull request merged**